### PR TITLE
UpgradeManager expose functions

### LINKED ISSUE #536: (Back) Have UpgradeManager expose functions
**Description :**
Have UpgradeManager expose functions instead of events as its base rather than feeding in eventemitter fns

**Acceptance Criteria :**
TO ADD

**Technical Notes:**
TO ADD

**Points:**
3pts

**Linked with**
https://github.com/digidem/mapeo-mobile/issues/538

### PR DESCRIPTION
**Pull request by hackergrrl**
Fixes #536

Recreated this from https://github.com/digidem/mapeo-mobile/pull/547 after I force-pushed the main `share-apk-p2p` dev branch.

### CODE REVIEW

**Comment by gmaclennan** [comment]
I offered to write the final integration for this (hooking up the events from upgradeManager to the nodejs RPC bridge and debouncing them), but before I do that I wanted to make sure @noffle is not working on the code in `src/backend/server.js` since this integration will touch a lot of the creation of the upgradeManager instance in that file.

**Comment by hackergrrl** [comment]
@luandro nice work!

@gmaclennan Please, go ahead

**Comment by gmaclennan** [comment]
@noffle this is ready for you to review. I could not get the upgrade-simulator to work - I tried `node ./bin/upgrade-simulator.js 5.5.0` and I think from the logs on the phone it was seeing the simulator, but it was not recognizing it as an upgrade. However I see the same behaviour on the `share-apk-p2p` branch.

**Comment by hackergrrl** [comment]
@gmaclennan The problem might be that the simulator is serving up the same APK as your phone is running, so the hashes match, and it ignores it. I've been getting around this (annoyingly) by commenting out all hash comparison logic in `lib/upgrade-download.js` while testing.

**Comment by gmaclennan** [comment]
Great, that was it. Working now.

You can build a version of the app with a different version number (and hence different hash) by:

```
cd android
ANDROID_VERSION_NAME=5.5.0 ./gradlew assembleAppDebug
```

Then move the apk from `android/app/build/outputs/apk/app/debug` somewhere before you run `npm run android`.

**Comment by gmaclennan** [comment]
@noffle I have merged the latest `share-apk-p2p` branch into this. This is ready to merge into `share-apk-p2p`. Would you like to review it more? I added what I think is a fix for the breaking tests (missing `fake2.apk` file). Please go ahead and merge if you think this is good to go.

File: src/backend/bin/upgrade-simulator.js [hunk 1]
```diff
@@ -14,37 +11,28 @@ console.log("tempdir:", dir);
 
 const apkPath =
   process.argv[3] ||
-  "../../android/app/build/outputs/apk/app/debug/mapeo-app-debug.apk";
+  path.join(
+    __dirname,
```
Comment (hackergrrl) [inline src/backend/bin/upgrade-simulator.js:15]: Nice catch!

**Review by hackergrrl — COMMENTED** [review]
Yay! Great work @luandro @gmaclennan! Looks good to me.

**Comment by hackergrrl** [comment]
>  I added what I think is a fix for the breaking tests (missing fake2.apk file)

Good catch! I totally forgot to commit that file.

**Pull request merged**